Mail/Hailgun/SendEmail.hs

```diff
@@ -54,13 +55,36 @@ toSimpleEmailParts message =
    ++ inReplyTo
    ++ references
    ++ fromContent (messageContent message)
+   ++ tags
+   ++ recipientVariables
+   ++ deliveryTime
+   ++ testMode
    where
       to = convertEmails (BC.pack "to") . messageTo $ message
       cc = convertEmails (BC.pack "cc") . messageCC $ message
       bcc = convertEmails (BC.pack "bcc") . messageBCC $ message
+      tags = map (\t -> (BC.pack "o:tag", T.encodeUtf8 t)) . messageTags $ message
+      recipientVariables =
+         if null (messageRecipientVariables message)
+         then []
+         else [(BC.pack "recipient-variables"
+              , BC.toStrict $ encode $ messageRecipientVariables message
+              )]
+
+      formatUTC :: UTCTime -> String
+      formatUTC = formatTime defaultTimeLocale "%a, %_d %b %Y %H:%M:%S %z"
+
+      deliveryTime = case messageDeliveryTime message of
+         Nothing -> []
+         Just t -> [(BC.pack "o:deliverytime", BC.pack $ formatUTC t)]
+
+      testMode = if messageTestMode message
+         then [(BC.pack "o:testmode", BC.pack $ "yes")]
+         else []
+
       replyTo = case messageReplyTo message of
          Nothing -> []
-         Just email -> [(BC.pack "h:reply-to", email)]
+         Just email -> [(BC.pack "h:Reply-to", email)]
```

Comment:
Did this matter or ?

Reply:
Im not 100% sure I was working through their docs and noticed they seems to be fairly specific about casing so figured it was best to be consistent